src/main/java/de/tum/in/www1/artemis/domain/Submission.java

```diff
@@ -110,12 +112,54 @@ public Result getLatestResult() {
         return null;
     }
 
+    /**
+     * Used to get result by correction round
+     * @param correctionRound to get result by
+     * @return the result based on the given correction-round
+     */
+    @Nullable
+    @JsonIgnore
+    public Result getResultByCorrectionRound(Long correctionRound) {
+        if (results != null && results.size() > correctionRound) {
+            return results.get(correctionRound.intValue());
+        }
+        return null;
+    }
+
+    /**
+     * Used to get result by correction round ignoring all automatic results
+     * @param correctionRound to get result by
+     * @return the result based on the given correction-round
+     */
+    @Nullable
+    @JsonIgnore
+    public Result getResultByCorrectionRoundIgnoreAutomatic(Long correctionRound) {
+        List<Result> withoutAutomaticResults = results.stream().filter(result -> result != null && !result.getAssessmentType().equals(AssessmentType.AUTOMATIC))
+                .collect(Collectors.toList());
+        if (withoutAutomaticResults != null && withoutAutomaticResults.size() > correctionRound) {
+            return withoutAutomaticResults.get(correctionRound.intValue());
+        }
+        return null;
+    }
+
     @Nullable
     @JsonProperty(value = "results", access = JsonProperty.Access.READ_ONLY)
     public List<Result> getResults() {
         return results;
     }
 
+    @Nullable
+    @JsonIgnore
+    public List<Result> getManualResults() {
+        return results.stream().filter(result -> result != null && !result.getAssessmentType().equals(AssessmentType.AUTOMATIC)).collect(Collectors.toList());
```

Comment:
can you explain more how you mean that, I don't really get where i should use the `!=`

Reply:
Sorry, I meant the comparison of enums. This one of the exceptions in Java where == is no problem and might make the code a bit cleaner. But this is not absolutely necessary, but would in this case also prevent a NPE in case `getAssessmentType` returns null. See also bottom of https://docs.oracle.com/javase/specs/jls/se15/html/jls-8.html#jls-8.9.1

Reply:
Or, if you use `equals`, call it on the contant, not the other way round. This also applies for String literals, and things like `Boolean.TRUE`. It allows to directly avoid any possible NPE without compromising readability.

Reply:
the equals needs to be the other way round to avoid null issues:
```suggestion
        return results.stream().filter(result -> result != null && !AssessmentType.AUTOMATIC.equals(result.getAssessmentType())).collect(Collectors.toList());
```